### PR TITLE
MultiQC, FastQC and bcftools stats implementation

### LINKED ISSUE #20: add multiqc rule to collect qc data
- Its a really useful tool https://multiqc.info/

- example usage in a snakemake workflow here  - https://github.com/sanjaynagi/rna-seq-pop/blob/master/workflow/rules/qc.smk

### PR DESCRIPTION
**Pull request by ChabbyTMD**
Resolves #20 
Added functionality for read and other metric quality control as well as MultiQC report generation.

### CODE REVIEW

**Comment by sanjaynagi** [comment]
Awesome, looks great Trevor!! 

Not sure why CI is failing - a quick google says it could be to do with the python version. Could try pinning to python to 3.7 in the ampseq_cli.yaml? We should actually be doing this anyway.

File: workflow/envs/AmpSeq_python.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ channels:
   - conda-forge
   - anaconda
 dependencies:
-  - python=3.9
+  - python=3.7
```
Comment (sanjaynagi) [inline workflow/envs/AmpSeq_python.yaml:7]:
```suggestion
  - python=3.9
```

**Comment by sanjaynagi** [comment]
Ahhh, didnt realise we had already set python 3.9 in the `Ampseq_python.yaml`. I think we should stick to 3.9 for that. 

The MultiQC rule will be using the python version that's install in `Ampseq_cli.yaml`, however, so we need to set the python version in there. Might make sense to also attempt using python=3.9 for that aswell. Have just done so, lets see if CI works

**Comment by sanjaynagi** [comment]
Ok ive tried to create the envs locally, and then using the mamba 'solved' yaml and using that instead in the workflow. This might end up being a bit of a pain when we want to add packages, but we shouldn't need to do that too often now.

**Comment by sanjaynagi** [comment]
I hate conda

**Comment by sanjaynagi** [comment]
hey @eddUG :)  Indeed, I've thought about doing this before, but occasionally you get wrappers which either don't work or are inflexible for what you need (i.e they don't allow some combinations of parameters). Although I agree, I suggest we just do it as and when we need to, as going through all the shell commands now and changing them will take some time.

**Comment by sanjaynagi** [comment]
@ChabbyTMD Let me know if youd like me to take this one :)

(and this is always an option!)

**Comment by ChabbyTMD** [comment]
Hi Sanjay, I can do the multiqc replacement with the wrapper. In the prototyping stage though I had issues using the fastqc wrapper. It would always error out at runtime not quite sure why. Using the shell command instead solved the issue.

File: workflow/rules/multiqc.smk [hunk 2]
```diff
@@ -15,6 +15,8 @@ rule FastQC:
         """
         fastqc {input} {params.outdir} -t {threads} 2> {log}
         """
+    # wrapper:
+    #     "v1.25.0/bio/fastqc"
```
Comment (ChabbyTMD) [inline workflow/rules/multiqc.smk:19]: Error at Runtime:
Error in rule FastQC:
    jobid: 19
    output: results/fastqc/sample97_1_fastqc.html, results/fastqc/sample98_1_fastqc.html, results/fastqc/sample97_2_fastqc.html, results/fastqc/sample98_2_fastqc.html, results/fastqc/sample97_1_fastqc.zip, results/fastqc/sample98_1_fastqc.zip, results/fastqc/sample97_2_fastqc.zip, results/fastqc/sample98_2_fastqc.zip
    log: logs/fastqc/fastq.log (check log file(s) for error message)
    conda-env: /home/tmugoya/PIPELINE/qc_branch/AmpSeeker/.snakemake/conda/94c868735e288e11b98ca93d754edd1d

RuleException:
CalledProcessError in line 19 of /home/tmugoya/PIPELINE/qc_branch/AmpSeeker/workflow/rules/multiqc.smk:
Command 'source /usr/local/anaconda/envs/snakemake/bin/activate '/home/tmugoya/PIPELINE/qc_branch/AmpSeeker/.snakemake/conda/94c868735e288e11b98ca93d754edd1d'; set -euo pipefail;  python /home/tmugoya/PIPELINE/qc_branch/AmpSeeker/.snakemake/scripts/tmpcdhxrfod.wrapper.py' returned non-zero exit status 1.
  File "/home/tmugoya/PIPELINE/qc_branch/AmpSeeker/workflow/rules/multiqc.smk", line 19, in __rule_FastQC
  File "/usr/local/anaconda/envs/snakemake/lib/python3.10/concurrent/futures/thread.py", line 58, in run

Comment (ChabbyTMD) [reply workflow/rules/multiqc.smk:19]: Fixed by using the shell command instead.

Comment (eddUG) [reply workflow/rules/multiqc.smk:19]: @ChabbyTMD Please share the content of logs/fastqc/fastq.log (if not too long) or attach the log file

**Comment by sanjaynagi** [comment]
Just tried a few things to get the conda working - moving fastqc into its own ampseeker-qc.yaml. 

FYI, where we use wrappers, there is no need to specify a conda env aswell.

**Pull request merged**